File: src/Ringor.Api/Models/ApplicationInfo.cs
using System;

namespace Dalion.Ringor.Api.Models {
    public class ApplicationInfo {
        public string Company { get; set; }
        public string Product { get; set; }
        public string Email { get; set; }
        public string Version { get; set; }
        public ApplicationUrlInfo UrlInfo { get; set; }
        public string Environment { get; set; }
        public ApplicationAuthenticationInfo AuthenticationInfo { get; set; }

        public class ApplicationUrlInfo {
            public string SiteUrl { get; set; }
            public string AppUrl { get; set; }
        }
        
        public class ApplicationAuthenticationInfo {
            public Uri Authority { get; set; }
            public string Tenant { get; set; }
            public string ClientId { get; set; }
            public string[] Scopes { get; set; }
        }
    }
}

File: src/Ringor/App/src/facades/applicationInfo.ts
export interface IWindow extends Window {
  applicationInfo: IApplicationInfo;
}
declare var window: IWindow;

export interface IUrlInfo {
  siteUrl: string;
  appUrl: string;
}

export interface IMsalConfig {
  authority: string;
  tenant: string;
  clientId: string;
  scopes: string[];
}

export interface IApplicationInfo {
  company: string;
  product: string;
  email: string;
  version: string;
  urlInfo: IUrlInfo;
  environment: string;
  msalConfig: IMsalConfig;
}

const applicationInfo = (typeof window !== "undefined" && window.applicationInfo || {}) as IApplicationInfo;

export default applicationInfo;